index.html

```diff
@@ -1,390 +1,1471 @@
+      the transmit buffer is important to maintain good throughput awaiting this
+      {{Promise}} before generating too many chunks of data is a good practice
+      to avoid excessive buffering.
+    </aside>
+
+    The {{SerialPort/writable}} getter steps are:
+
+    <ol>
+      <li>
+        If [=this=].{{[[writable]]}} is not `null`, return
+        [=this=].{{[[writable]]}}.
+      <li>If [=this=].{{[[state]]}} is not `"opened"`, return `null`.
+      <li>If [=this=].{{[[writeFatal]]}} is `true`, return `null`.
+      <li>Let |stream:WritableStream| be a [=new=] {{WritableStream}}.
+      <li>Let |writeAlgorithm| be the following steps, given |chunk|:
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          <li>
+            If |chunk| cannot be [=converted to an IDL value=] of type
+            {{BufferSource}}, reject |promise| with a {{TypeError}} and return
+            |promise|. Otherwise, save the result in |source:BufferSource|.
+          <li>
+            [=Get a copy of the buffer source=] |source| and save the result in
+            |bytes|.
+          <li>
+            [=In parallel=], run the following steps:
+            <ol>
+              <li>
+                Invoke the operating system to write |bytes| to the port.
+                <div class="note">
+                  The operating system may return from this operation once
+                  |bytes| has been queued for transmission rather than after it
+                  has been transmitted.
+                </div>
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=serial port task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    If the chunk was successfully written [=resolve=] |promise|
+                    with `undefined`.
+                  <li>
+                    If an operating system error was encountered, [=reject=]
+                    |promise| with an "{{UnknownError}}" {{DOMException}}.
+                  <li>
+                    If the port was disconnected, run the following steps:
+                    <ol>
+                      <li>Set [=this=].{{[[writeFatal]]}} to `true`.
+                      <li>
+                        [=Reject=] |promise| with a "{{NetworkError}}"
+                        {{DOMException}}.
+                      <li>
+                        Invoke the steps to [=handle closing the writable
+                        stream=].
+                    </ol>
+                </ol>
+            </ol>
+          <li>Return |promise|.
+        </ol>
+
+        <div class="note">
+          [[STREAMS]] specifies that |writeAlgorithm| will only be invoked after
+          the {{Promise}} returned by a previous invocation of this algorithm
+          has resolved. For efficiency an implementation MAY coalesce multiple
```

Comment:
"MAY" requirements should not be in non-normative notes. This should probably be a normative modification to step 5.4.1, something like "Invoke the operating system to write _bytes_ to port. Alternately, store the chunk for future coalescing." Then maybe step 5.4.2.1 says "If the chunk was successfully written, or was stored for future coaelescing, resolve promise with undefined".